File: android/AgilityFitTodayApp/app/src/main/java/com/laurenyew/agilityfittodayapp/data/models/WorkoutType.kt
package com.laurenyew.agilityfittodayapp.data.models

import com.squareup.moshi.FromJson
import com.squareup.moshi.JsonAdapter
import com.squareup.moshi.JsonReader

enum class WorkoutType {
    CARDIO,
    CORE,
    UPPER_BODY_STRENGTH,
    LOWER_BODY_STRENGTH,
    STRETCH,
    STRENGTH,
    REST;

    override fun toString(): String =
        when (this) {
            UPPER_BODY_STRENGTH -> "Upper-body Strength"
            LOWER_BODY_STRENGTH -> "Lower-body Strength"
            else -> name.toLowerCase().capitalize()
        }
}

class WorkoutTypeAdapter {
    private val workoutTypeValues = WorkoutType.values()

    @FromJson
    fun fromJson(jsonReader: JsonReader, delegate: JsonAdapter<WorkoutType>): WorkoutType {
        val value = jsonReader.nextInt()
        return workoutTypeValues[value]
    }
}

File: ios/AgilityFitTodayApp/AgilityFitTodayApp/Models/WorkoutType.swift
//
//  WorkoutType.swift
//  AgilityFitTodayApp
//
//  Created by laurenyew on 3/30/22.
//

import Foundation

enum WorkoutType: String, Codable {
    case Cardio = "CARDIO"
    case Core = "CORE"
    case UpperBodyStrength = "UPPER_BODY_STRENGTH"
    case LowerBodyStrength = "LOWER_BODY_STRENGTH"
    case Stretch = "STRETCH"
    case Strength = "STRENGTH"
    case Rest = "REST"
    
    func uiString() -> String {
        switch(self){
        case .UpperBodyStrength:
            return "Upper-body Strength"
        case .LowerBodyStrength:
            return "Lower-body Strength"
        default:
            return self.rawValue.capitalized(with: Locale.current)
        }
    }
}

extension WorkoutType: Identifiable {
    var id: Self { self }
}
